### PR TITLE
Automatically validate and publish integration test results into repo

### PR DESCRIPTION
**Pull request by robbibt**
This PR replicates the [automated validation test results we have for DEA Coastlines](https://github.com/GeoscienceAustralia/dea-coastlines/tree/develop/tests) on the DEA Intertidal repo. This:

- Runs an integration test using the Intertidal CLI
- Compares output elevation results to a validation LiDAR DEM and calculates some accuracy stats
- Automatically generates a README markdown and plot containing a summary of the results (and how they compare to previous runs)
- Commits those files back into the repo so we can easily check whether changes to the code have affected accuracy.

Preview of automatically generated README: https://github.com/GeoscienceAustralia/dea-intertidal/tree/tests/tests

image

### CODE REVIEW

File: .github/workflows/dea-intertidal-image.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ on:
 
 permissions:
   id-token: write   # This is required for requesting the JWT
-  contents: read    # This is required for actions/checkout
+  contents: write    # This is required for actions/checkout
```
Comment (robbibt) [inline .github/workflows/dea-intertidal-image.yml:28]: Needed to commit back into repo

File: .github/workflows/dea-intertidal-image.yml [hunk 2]
```diff
@@ -52,14 +52,27 @@ jobs:
           # Run integration tests using Docker
           docker compose run dea_intertidal pytest -v --cov=intertidal --cov-report=xml tests
 
+          # Copy out validation outputs produced by the integration tests and place them
```
Comment (robbibt) [inline .github/workflows/dea-intertidal-image.yml:55]: We can't commit directly inside the container, so the files need to first be copied from the mounted "artifacts" directory, then copied into their correct locations on the main Github runner (outside of the container). We can then commit them via Git in the next step

File: codecov.yaml [hunk 3]
```diff
@@ -1,5 +1,4 @@
 ignore:
-  - "intertidal/validation.py"
```
Comment (robbibt) [inline codecov.yaml:2]: Stop ignoring validation code as we now cover some of it in our validation tests

File: docker-compose.yml [hunk 4]
```diff
@@ -32,6 +32,7 @@ services:
     volumes:
       - .:/code
       - ./tide_models/:/var/share/tide_models
+      - ./artifacts:/mnt/artifacts
```
Comment (robbibt) [inline docker-compose.yml:35]: Mounted volume to let us copy files out of the container

File: requirements.in [hunk 5]
```diff
@@ -6,6 +6,7 @@ datacube[s3,performance]==1.8.13
 dea-tools==0.3.1.dev5
 geopandas==0.13.2
 matplotlib==3.7.1
+mdutils
```
Comment (robbibt) [inline requirements.in:9]: Needed to auto-generate markdown readme

File: tests/test_intertidal.py [hunk 6]
```diff
@@ -23,11 +42,204 @@ def test_intertidal_cli():
...
             "3h",
+            # "--windows_n",
+            # 50,
         ],
     )
     assert result.exit_code == 0
 
 
+@pytest.mark.dependency(depends=["test_intertidal_cli"])
+def test_dem_accuracy(
```
Comment (robbibt) [inline tests/test_intertidal.py:53]: Function that compares integration test results to LiDAR validation, and exports plot and readme summary files

**Review by vnewey — APPROVED** [review]
Awesome work Robbi!

**Pull request merged**